core-web/libs/sdk/experiments/src/lib/hooks/useExperiments.ts

```diff
@@ -0,0 +1,78 @@
+// Update the location of the DotExperiments instance when it changes
+import { useContext, useEffect } from 'react';
+
+import { isInsideEditor } from '@dotcms/client';
+
+import DotExperimentsContext from './DotExperimentsContext';
+
+import { EXPERIMENT_DEFAULT_VARIANT_NAME, EXPERIMENT_QUERY_PARAM_KEY } from '../shared/constants';
+
+/**
+ * `useExperiments` is a custom hook for updating the location of the DotExperiments instance when it changes.
+ *
+ * @returns {void}
+ */
+export const useExperiments = (): void => {
+    // retrieve experiment context
+    const experimentContext = useContext(DotExperimentsContext);
+    /**
+     * This effect adds a change listener to the experiment context, specifically updating the location.
+     * It doesn't have a meaningful return; it primarily triggers side-effects.
+     */
+    useEffect(() => {
+        if (!experimentContext || typeof document === 'undefined') {
+            return;
+        }
+
+        const insideEditor = isInsideEditor();
+        if (!insideEditor) {
+            const location = typeof window !== 'undefined' ? window.location : undefined;
+            if (experimentContext && location) {
+                experimentContext.locationChanged(location, experimentContext.customRedirectFn);
+            }
+        }
+    }, [experimentContext]);
+
+    /**
+     * This effect sets a click handler on the document.
+     * It captures click events and redirects to a new URL if the clicked anchor has a variant assigned.
+     */
+    useEffect(() => {
+        if (!experimentContext) {
+            return;
+        }
+
+        const customClickHandler = (event: MouseEvent) => {
+            const target = event.target as HTMLAnchorElement;
+            // Check if the clicked element is an anchor
+            if (target.tagName.toLowerCase() === 'a') {
+                const clickedHref = target.getAttribute('href');
```

Comment:
If you have like `<a><span></span></a>` this will not caught.

Reply:
let me try

Reply:
Handled.